src/Authenticator.ts

```diff
@@ -3,63 +3,117 @@ import Mustache = require('mustache');
 import readPkg = require('read-pkg');
 import path = require('upath');
 
-const Zip = require('node-zip');                   // tslint:disable-line:no-var-requires  no type definitions available
+const Zip = require('node-zip'); // tslint:disable-line:no-var-requires  no type definitions available
 import { coerce, SemVer } from 'semver';
 import { ensure, isArray, isGreaterThan, isString, property } from 'tiny-types';
 
 export class Authenticator {
-    /**
-     * @param {string} username
-     * @param {string} password
-     * @param {string[]} permissions
-     *  See https://developer.chrome.com/extensions/declare_permissions
-     */
-    static for(username: string, password: string, permissions: string[] = ['<all_urls>']): Authenticator {
-        return new Authenticator(username, password, permissions);
-    }
+  /**
+   * @param {string} username
+   * @param {string} password
+   * @param {string[]} permissions
+   *  See https://developer.chrome.com/extensions/declare_permissions
+   */
+  static for(
+    username: string,
+    password: string,
+    permissions: string[] = ['<all_urls>']
+  ): Authenticator {
+    return new Authenticator(username, password, permissions);
+  }
 
-    asBase64(): string {
-        return this.extension().generate({ base64: true, compression: 'DEFLATE' });
-    }
+  asBase64(): string {
+    return this.extension().generate({ base64: true, compression: 'DEFLATE' });
+  }
+
+  asFileAt(_path: string): string {
+    const zip = new Zip(
+      this.extension().generate({
+        base64: false,
+        compression: 'DEFLATE',
+      }),
+      {
+        base64: false,
+        checkCRC32: true,
+      }
+    );
 
-    private constructor(
-        private readonly username: string,
-        private readonly password: string,
-        private readonly permissions: string[],
-    ) {
-        ensure('username', username, isString(), property('length', isGreaterThan(0)));
-        ensure('password', password, isString(), property('length', isGreaterThan(0)));
-        ensure('permissions', permissions, isArray(), property('length', isGreaterThan(0)));
+    if (!fs.existsSync(_path)) {
+      fs.mkdirSync(_path);
     }
+    fs.writeFileSync(
+      path.join(_path, 'manifest.json'),
+      zip.files['manifest.json']._data
+    );
+    fs.writeFileSync(
+      path.join(_path, 'authenticator.js'),
+      zip.files[`authenticator.js`]._data
+    );
+    return _path;
+  }
```

Comment:
Could we not reuse `this.extension().generate({ base64: true, compression: 'DEFLATE' })`?

Reply:
Puppeteer needs physical files to be present at the specified directory. I have used ```this.extension().generate({ base64: true, compression: 'DEFLATE' })``` to unzip the file. 

```
const zip = new Zip(
            this.extension().generate({
                base64: false,
                compression: 'DEFLATE',
            }),
            {
                base64: false,
                checkCRC32: true,
            }
        );

```